src/expound/util.cljc

```diff
@@ -5,3 +5,9 @@
 (defn nan? [x]
   #?(:clj (and (number? x) (Double/isNaN x))
      :cljs (and (number? x) (js/isNaN x))))
+
+(defmacro priml [x]
+  #?(:clj `(long ~x)
```

Comment:
I might be wrong, but doesn't the `:cljs` branch only run for self-compiled cljs?

i.e. a `long` call will be emitted for vanilla cljs, because defmacro always run via the JVM for vanilla cljs, so the `:clj` reader conditional is always the picked branch there.

For that kind of reason I tend to detect the compilation target like this: `(-> &env :ns nil?) ;; true for JVM clj, false for cljs`

Reply:
That seems plausible, but then I'm still wondering why the CLJS specs all pass? Either

1. The specs are not hitting this code (seems implausible)
2. I'm misunderstanding something about the combo of reader conditionals and macros (I tend to agree with your analysis, but I haven't dug in)
3. It's fine to call `(long x)` in CLJS

In my brief research, it looks like it might be (3)? https://cljs.github.io/api/cljs.core/#long

There may be a downside I'm unaware of, but if that's correct, I'm leaning towards just removing this macro and the reader conditional and just replacing it with `long`

Reply:
For clarity - I'm sure that my analysis is correct (as I have long-tested macros which exercise JVM + cljs), but I said `I might be wrong` mostly for acknowledging that other approaches to macro writing are also feasible.

So yes, as of the current implementation `long` is emitted for vanilla cljs , and not for self-hosted cjls; and the extra `long` call isn't tremendously problematic.

At the same time, the `long` has no benefit in cljs, and in fact it worsens peformance as it's one extra function call in the stack. Also a bit fragile to have different behaviors between two clojurescripts.

I'd suggest - either tweak the macro to emit no `long` call in any of the two possible clojurescript targets, or use my original implementation which uses no macros, at the cost of a slightly repetitive pattern.

Reply:
Thanks for all the information. It's greatly appreciated!

I appreciate the input, but I think I'm just going to do the simplest thing here and add a `long` call instead the macro or reader conditionals. I suspect the extra function call has negligible performance impacts and in any case, Expound is only called in error situations, so unlikely to be called in performance-sensitive code.